tests/integration/support/steps.go

```diff
@@ -0,0 +1,271 @@
+package support
+
+import (
+	"encoding/json"
+	"fmt"
+	"regexp"
+	"strconv"
+	"strings"
+
+	"github.com/optimizely/go-sdk/pkg/event"
+
+	"github.com/DATA-DOG/godog/gherkin"
+	"github.com/facebookarchive/subset"
+	"github.com/optimizely/go-sdk/pkg/entities"
+	"github.com/optimizely/go-sdk/tests/integration/models"
+	"github.com/optimizely/go-sdk/tests/integration/optlyplugins"
+	"gopkg.in/yaml.v3"
+)
+
+// ScenarioCtx holds both apiOptions and apiResponse for a scenario
+type ScenarioCtx struct {
+	apiOptions  models.APIOptions
+	apiResponse models.APIResponse
+	wrapper     ClientWrapper
+}
+
+// TheDatafileIs represents a step in the feature file
+func (c *ScenarioCtx) TheDatafileIs(datafileName string) error {
+	c.wrapper = NewWrapper(datafileName)
+	return nil
+}
+
+// ListenerIsAdded represents a step in the feature file
+func (c *ScenarioCtx) ListenerIsAdded(numberOfListeners int, ListenerName string) error {
+	if c.apiOptions.Listeners == nil {
+		c.apiOptions.Listeners = make(map[string]int)
+	}
+	c.apiOptions.Listeners[ListenerName] = numberOfListeners
+	return nil
+}
+
+// IsCalledWithArguments represents a step in the feature file
+func (c *ScenarioCtx) IsCalledWithArguments(apiName string, arguments *gherkin.DocString) error {
+	c.apiOptions.APIName = apiName
+	c.apiOptions.Arguments = arguments.Content
+	result, err := c.wrapper.InvokeAPI(c.apiOptions)
+	//Reset listeners so that same listener is not added twice for a scenario
+	c.apiOptions.Listeners = nil
+	if err == nil {
+		c.apiResponse.Result = result.Result
```

Comment:
Why don't we set `c.apiResponse` to `c.clientWrapper.InvokeAPI` and return the err if err is not nil?